Neuro/Pathfinding/PathfindingThread.cs

```diff
@@ -274,30 +284,25 @@ private static Vector2[] RetracePath(Node startNode, Node endNode)
             currentNode = currentNode.parent;
         }
 
-        Vector2[] waypoints = path.Select(p => p.worldPosition).ToArray();
-        new Span<Vector2>(waypoints).Reverse();
-
-        return SimplifyPath(waypoints);
+        path.Reverse();
+        return SimplifyPath(path);
     }
 
-    static Vector2[] SimplifyPath(Vector2[] path)
+    private static Vector2[] SimplifyPath(List<Node> path)
     {
-        List<Vector2> waypoints = new();
+        List<Vector2> waypoints = new() {path[0].worldPosition};
         Vector2 directionOld = Vector2.zero;
-        for (int i = 1; i < path.Length; i++)
+        for (int i = 1; i < path.Count; i++)
         {
-            Vector2 directionNew = new Vector2(path[i - 1].x - path[i].x, path[i - 1].y - path[i].y);
-            if (directionNew != directionOld)
+            Vector2 directionNew = new(path[i - 1].worldPosition.x - path[i].worldPosition.x, path[i - 1].worldPosition.y - path[i].worldPosition.y);
+            if (directionNew != directionOld || path[i].transportSelfId != 0)
             {
-                waypoints.Add(path[i]);
+                waypoints.Add(path[i].worldPosition);
             }
 
             directionOld = directionNew;
         }
 
-        // add last waypoint
-        waypoints.Add(path[^1]);
```

Comment:
Is there a specific reason this was removed? I believe it was there to prevent IOOB when directly next to the task
```
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at Neuro.Pathfinding.PathfindingThread.SimplifyPath(List`1 path) in C:\Users\foo\source\repos\neuro-amongus\Neuro\Pathfinding\PathfindingThread.cs:line 293
   at Neuro.Pathfinding.PathfindingThread.RetracePath(Node startNode, Node endNode) in C:\Users\foo\source\repos\neuro-amongus\Neuro\Pathfinding\PathfindingThread.cs:line 288
   at Neuro.Pathfinding.PathfindingThread.FindPath(Vector2 start, Vector2 target) in C:\Users\foo\source\repos\neuro-amongus\Neuro\Pathfinding\PathfindingThread.cs:line 146
   at Neuro.Pathfinding.PathfindingThread.RunThread() in C:\Users\foo\source\repos\neuro-amongus\Neuro\Pathfinding\PathfindingThread.cs:line 72
```